src/test/shell/bazel/bazel_cc_code_coverage_test.sh

```diff
@@ -0,0 +1,139 @@
+#!/bin/bash -eu
+#
+# Copyright 2018 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Unit tests for tools/test/collect_cc_code_coverage.sh
+
+
+# Load the test setup defined in the parent directory
+CURRENT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+source "${CURRENT_DIR}/../integration_test_setup.sh" \
+  || { echo "integration_test_setup.sh not found!" >&2; exit 1; }
+
+function setup_cc_sources() {
+  mkdir -p coverage_srcs/
+  cat << EOF > coverage_srcs/a.h
+int a(bool what);
+EOF
+
+  cat << EOF > coverage_srcs/a.cc
+#include "a.h"
+
+int a(bool what) {
+  if (what) {
+    return 1;
+  } else {
+    return 2;
+  }
+}
+EOF
+
+  cat << EOF > coverage_srcs/t.cc
+#include <stdio.h>
+#include "a.h"
+
+int main(void) {
+  a(true);
+}
+EOF
+}
+
+function generate_gcc_and_gcda_files() {
+  # Generate .gcno files.
+  g++ -fprofile-arcs -ftest-coverage -c coverage_srcs/a.h coverage_srcs/a.cc coverage_srcs/t.cc
+  # Produce instrumented binary.
+  g++ -fprofile-arcs -ftest-coverage coverage_srcs/a.h coverage_srcs/a.cc coverage_srcs/t.cc -o ./coverage_srcs/test
+  # Execute the test and generate test.gcda file.
+  ./coverage_srcs/test
+}
+
+function set_up() {
+  check_env
+  export COVERAGE_DIR="$PWD"
```

Comment:
If you're going to export these variables please do it in a subshell.

Example:
(COVERAGE_DIR="$PWD" my_command)

Reply:
Also, if these are global readonly variables you can define them at the beginning of the script as:
readonly COVERAGE_DIR="$PWD"

Reply:
Done.